blocks/blog-filter/blog-filter.js

```diff
@@ -0,0 +1,189 @@
+    image.alt = post.title;
+    itemLeft.append(image);
+    image.src = post.image;
+    item.append(itemLeft);
+    itemRight.append(heading, categoryDiv, dateDiv);
+    item.append(itemRight);
+    list.append(item);
+  });
+}
+
+async function updateResults(checkboxValue) {
+  const postArray = [];
+  const posts = await ffetch('/query-index.json').sheet('blog')
+    .map((post) => ({
+      tags: post.tags.split(',').map((tag) => tag.trim().replaceAll(/["[\]]/g, '')),
+      title: post.title, // Include the title
+      date: post.date, // Include the date
+      image: post.image, // Include the image
+    }))
+    .filter((post) => post.tags.includes(checkboxValue))
+    .limit(10)
+    .all();
+  posts.forEach((post) => {
+    postArray.push(post);
+  });
+  const bfc = document.querySelector('div.blog-filter-container> div.results> ul');
+  bfc.innerHTML = '';
+  decorateResults(postArray, bfc);
+
+  const checkboxes = document.querySelectorAll('div.blog-filter-container div.facet input[type="checkbox"]');
+  checkboxes.forEach((checkbox) => {
+    if (checkbox.value !== checkboxValue) {
+      checkbox.checked = false;
+    }
+  });
+}
+
+function clickChevron() {
+  const fl = document.querySelector('ul.facet-list');
+  if (fl.classList.contains('slideout')) {
+    fl.classList.remove('slideout');
+  } else {
+    fl.classList.add('slideout');
+  }
+}
+
+export default async function decorate(block) {
+  const facets = await getFacets();
+  const ph = await fetchPlaceholders(`/${getLocale()}`);
+  const { blogtopic } = ph;
+  const facetDiv = document.createElement('div');
+  const mobileFilter = document.createElement('i');
+  mobileFilter.classList.add('filter-icon');
+  mobileFilter.addEventListener('click', () => {
+    toggleFacets();
+  });
+  facetDiv.append(mobileFilter);
+  facetDiv.classList.add('facet');
+  const facetList = document.createElement('ul');
+  const facetLI = document.createElement('li');
+  const topicHead = document.createElement('div');
+  topicHead.classList.add('topic-head');
+  const chevron = document.createElement('i');
+  chevron.classList.add('fa-chevron-down');
+  chevron.addEventListener('click', () => {
+    clickChevron();
+  });
+  topicHead.innerText = blogtopic;
+  topicHead.append(chevron);
+  facetLI.append(topicHead);
+  facetList.append(facetLI);
+  const facetUL = document.createElement('ul');
+  facetUL.classList.add('facet-list');
+  facetLI.append(facetUL);
+  facets.tags.forEach((facet) => {
+    const facetItem = document.createElement('li');
+    const topic = document.createElement('div');
+    topic.classList.add('facet-group');
+    const checkbox = document.createElement('input');
+    const label = document.createElement('label');
+    label.innerText = `${facet.tag} (${facet.count})`;
+    checkbox.type = 'checkbox';
+    checkbox.value = facet.tag;
+    // create a listener for the checkbox
+    checkbox.addEventListener('change', (cb) => {
```

Comment:
Not sure why, but for me it's only updating the results when checking a bock, not when unchecking one

Reply:
I think I know why. I'm passing the checkbox value and not the element - so it can't check for if its 'checked' or not.